src/Domain/Tags/ActiveMesh.hpp

```diff
@@ -0,0 +1,40 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain::Tags {
+/// \cond
+template <size_t VolumeDim>
+class Mesh;
+/// \endcond
+
+/// \brief The mesh of the active computational grid. For hybrid methods like
+/// DG-FD the active mesh changes throughout the evolution.
+template <size_t Dim>
+struct ActiveMesh : db::SimpleTag {
```

Comment:
Can you move this to evolution/subcell? It says subcell in the dox.

Also, could you explain your plan for this tag a bit? This being in domain makes me worried that code that has nothing to do with subcell (like elliptic, or things like horizon finders, etc -- meshes are used everywhere!) has to deal with this.

Reply:
Great question! No, the reason for this design is actually to simplify the observers. Currently I maintain IO events just for subcell, which are basically the same as the DG+subcell ones, and so adds a lot of maintenance overhead. The only differences are 1) the mesh being used has to be determined (either DG or subcell, they are stored separately in the box) 2) the coordinates (again stored separately). The `Active*` tags will only be used in observing, and the subcell-specific compute tags will be in the subcell directory. I'm intentionally not modifying `domain::Tags::Mesh` to minimize intrusion into the other systems. I view this change to be somewhat similar to the addition of the `SectionId` to the observers, only needed by some executables. The general change needed to all executables is that they add `ActiveMeshCompute` to the list of non-tensor compute tags in ObserveFields. The designs Larry, Mark, and I considered are:
1. Maintaining an additional event (not great because it's like 1200+ lines of almost copy-paste, even with code reuse)
2. Adding template parameters for `Coordinates` and `Mesh` to the `Observe*` events. This just causes an explosion of template parameters one needs to get right and so isn't very user-friendly. It's also unclear the new template parameters would be good to default. I would say _most_ executables will use DG-subcell, and so default to just DG is bad/not intuitive.
3. The design here.

Reply:
Thanks for explaining. Removing the duplicate subcell observation events sounds great, I'm all for it. From #3766 it looks like the toggle to use the correct mesh/coords is the only thing that's needed, so that's great. The only thing that I'm not super happy about is the addition of `domain::Tags::ActiveMeshFromMeshCompute<volume_dim>` in executables that have nothing to do with subcell. It seems like another thing to explain to a person in a "how to write an executable" tutorial, complicating the metavars a bit further. I would quite strongly prefer your option 2. here, adding the template parameter to `dg::Events::field_observations`. That's how `TimeTag` and `SectionId` are already handled, and replacing template parameters with compute items doesn't seem better to me. The observe events are "templates" in the sense that they are written to handle all kinds of executables to reduce code duplication. I agree that no default for the `MeshTag` template param is needed.

Reply:
No, I don't think that makes the code any easier to deal with. I very much prefer the current design as opposed to adding yet more template parameters. As I said, that was the result of discussions with Mark and Larry too... With template parameters you _still_ need to explain them in the tutorial, nothing has changed there. The only difference is that now every observing-type event will need 2+ more template parameters. Considering 50% or more of our executables do subcell, I don't think it's reasonable to treat subcell as the "exception", it's the norm, actually.

Reply:
I'm also confused on something. You're saying having an explicit template parameter is fine. This would mean that you have `observe_fields<..., domain::Tags::ActiveMeshCompute>`. The current design is `observe_fileds<..., tmpl::list<..., domain::Tags::ActiveMeshCompute>>`. Am I understanding what you're saying correctly? If so, how does moving the template parameter from inside the list to outside make a big difference to you?

Reply:
Just to answer your question: It would be `observe_fields<..., domain::Tags::Mesh<Dim>>` for executables that have only one mesh. That's quite unsurprising for a person reading the code, whereas an additional `domain::Tags::ActiveMeshFromMeshCompute<volume_dim>` compute tag would need some explanation.